### ISSUE TITLE
关于Matrix dimensions are not equal的问题

### ISSUE THREAD

**Issue by KaelChen** [PaddlePaddle/Paddle#7456]
构造data_layer的代码如下：
```python
X = paddle.layer.data(name="X", type=paddle.data_type.dense_vector_sequence(12))
```
生成数据的代码如下：
```python
data = np.array([[0.] * 12] * 46)
yield data
```
然而实际执行的时候报错：Check failed: height_ == label.getSize() && width_ == 1 && height_ == output.height_ Matrix dimensions are not equal
堆栈信息上看似乎是bp的时候出错的。
堆栈信息如下：
*** Check failure stack trace: ***
    @     0x7f95cda4e5ed  google::LogMessage::Fail()
    @     0x7f95cda5209c  google::LogMessage::SendToLog()
    @     0x7f95cda4e0e3  google::LogMessage::Flush()
    @     0x7f95cda535ae  google::LogMessageFatal::~LogMessageFatal()
    @     0x7f95cd87f482  paddle::GpuMatrix::oneHotCrossEntropy()
    @     0x7f95cd738df5  paddle::CostLayer::forward()
    @     0x7f95cd773269  paddle::NeuralNetwork::forward()
    @     0x7f95cd773ee3  paddle::GradientMachine::forwardBackward()
    @     0x7f95cda2a9e9  GradientMachine::forwardBackward()
    @     0x7f95cd62924c  _wrap_GradientMachine_forwardBackward
    @           0x4b4cb9  PyEval_EvalFrameEx
    @           0x4b6b28  PyEval_EvalCodeEx
    @           0x4b5d10  PyEval_EvalFrameEx
    @           0x4b6b28  PyEval_EvalCodeEx
    @           0x4b5d10  PyEval_EvalFrameEx
    @           0x4b6b28  PyEval_EvalCodeEx
    @           0x4b5d10  PyEval_EvalFrameEx
    @           0x4b6b28  PyEval_EvalCodeEx
    @           0x4b5d10  PyEval_EvalFrameEx
    @           0x4b5fb8  PyEval_EvalFrameEx
    @           0x4b6b28  PyEval_EvalCodeEx
    @           0x4b6c52  PyEval_EvalCode
    @           0x4e1c7d  PyRun_FileExFlags
    @           0x4e3501  PyRun_SimpleFileExFlags
    @           0x4159dd  Py_Main
    @     0x7f960459bbd5  __libc_start_main
    @           0x414b71  (unknown)
    @              (nil)  (unknown)

麻烦大神帮忙看一下，谢谢。

**Comment by ranqiu92** [comment]
提供下完整的代码，看报错 这个可能也与其他layer有关

**Comment by KaelChen** [comment]
附上网络定义代码：
```python
def __rnn_net__(self, layer, networkid):
    for i in xrange(2):
        layer = self.__rnn_layer__(layer, i, networkid)
    for i in xrange(2):
        layer = self.__dense_bn_layer__(layer, i)
    return layer

def __rnn_layer__(self, layer, i, networkid):
    name = 'network_%d_layer_%d' %(networkid, i)
    reverse = True if i % 2 == 1 else False
    def __rnn_cell__(input_layer):
        memory_layer = paddle.layer.memory(name=name, size=128)
        cell = paddle.layer.mixed(
                                  input=[paddle.layer.full_matrix_projection(input=input_layer, 
                                  param_attr=paddle.attr.Param(name='rnn_input_param_%d' %i)), 
                                  paddle.layer.full_matrix_projection(input=memory_layer, 
                                  param_attr=paddle.attr.Param(name='rnn_mem_param_%d' %i))], 
                                  name=name, size=128, act=paddle.activation.Relu())
        return cell
    return paddle.layer.recurrent_group(name='%s_recurrent_group' %name, 
               step=__rnn_cell__, reverse=reverse, input=layer)

def __call__(self, X):
    layers = []
    for i in xrange(2):
        layers.append(self.__rnn_net__(X[i], i))
    layer = paddle.layer.concat(layers)
    logits = paddle.layer.fc(input=layer, size=2, act=paddle.activation.Softmax())
```

上述代码中每一个```X[i]``` 都是上面定义的 ```X```

**Comment by KaelChen** [comment]
以下是cost layer及其他部分代码：
```python
paddle.init(use_gpu=params.use_gpu, trainer_count=params.trainer_count, gpu_id=params.gpu_id)
X_0 = build_input(params, "X_0")
X_1 = build_input(params, "X_1")
y = paddle.layer.data(name='y', type=paddle.data_type.integer_value(2))
ltr_net = network.RouteLtrNet(params)
predict = ltr_net([X_0, X_1])
cost = paddle.layer.classification_cost(input=predict, label=y)
parameters = paddle.parameters.create(cost)
optimizer = paddle.optimizer.Adam(learning_rate=params.learning_rate/params.batch_size)
trainer = paddle.trainer.SGD(cost=cost, parameters=parameters, update_equation=optimizer, 
extra_layers=paddle.evaluator.auc(input=predict, label=y))
```
其中```ltr_net``` 对应上面的 ```__call__``` 方法

**Comment by ranqiu92** [comment]
这里没有__dense_bn_layer__。
据推测，你的网络softmax输出是一个seq，元素个数和输入的seq的元素个数相同，而label也就是y的元素个数为1 ，  个数不等，所以就报了这个错。

**Comment by KaelChen** [comment]
加入全连接层代码：
```python
def __dense_bn_layer__(self, input_layer, i):
    layer = paddle.layer.fc(input=input_layer,
                            size=1,
                            act=None,
                            bias_attr=False,
                            param_attr=paddle.attr.Param(name="fc_layer_%d" %i))
    layer = paddle.layer.batch_norm(input=layer,
                            batch_norm_type="cudnn_batch_norm",
                            act=paddle.activation.Relu(),
                            use_global_stats=True,
                            param_attr=paddle.attr.Param(name="fc_bn_%d" %i))
    return layer
```

我的rnn输出结果进入了全连接层，最终应当输出一个1维的标量才对，为什么输出还是序列呢？

**Comment by ranqiu92** [comment]
全连接层 是分别对seq中的每个元素进行运算，其输出不改变seq的元素个数

**Comment by peterzhang2029** [comment]
Closing due to low activity. Feel free to reopen it.
